libcloud.groovy

```diff
@@ -46,10 +48,89 @@ def replicate_to_clouds(pipecfg, basearch, buildID, stream) {
             """)
         }
     }
+    // A closure to build the aws-winli AMI. This will be called before replicating to 
+    // all regions, if the option is set for the current stream. `cosa aws-replicate` 
+    // will handle both traditional AMIs and aws-winli AMIs if present in the metadata. 
+    // aws-winli is only supported on x86_64.
+    def awsWinLIBuildClosure = { config, aws_image_name, credentialId ->
+        def creds = [file(variable: "AWS_CONFIG_FILE", credentialsId: credentialId)]
+        withCredentials(creds) {
+            utils.syncCredentialsIfInRemoteSession(["AWS_CONFIG_FILE"])
+            def c = config
+
+            // Since we are not uploading anything, let's just touch the vmdk image
+            // file to satisfy ore aws upload, which still requires the file in the
+            // cosa working dir.
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, stream)
+            shwrap("""
+                touch builds/${buildID}/${basearch}/${aws_image_name}
```

Comment:
Should we clean this up when we're done running `cosa imageupload-aws`? Not sure if it might cause any issues (i.e. wouldn't want this empty file to get uploaded back overtop the actual file somehow).

Reply:
That's a good idea. I updated the PR to include a `rm` command on it. I dont think it would be uploaded since this is the replicate portion of the code, but it's better to be safe!